### DISCUSSION TITLE
Product and p2.inf questions

### DISCUSSION THREAD

**Discussion by andrew-tram** [eclipse-pde/eclipse.pde#1155] [General]
When creating an RCP, what's the point of having the iu included in the p2.inf (requires.1.namespace, requires.1.name, requires.1.filter) when the features/ plugins are already included in the product file?

Example:

In product file...

`<feature id="org.eclipse.platform"/>`

In p2.inf...

```
requires.102.namespace = org.eclipse.equinox.p2.iu
requires.102.name = org.eclipse.platform.feature.group
requires.102.filter = (org.eclipse.epp.install.roots=true)
```

Also, where does this string come from `org.eclipse.epp.install.roots` ? Let's say my product id is `my.company.product`, would I use `my.company.product.install.roots=true` or someting?

**Comment by merks** [comment]
This is quite specific to EPP packages, though it's a general issue.  For EPP this is done for feature that use `installMode="root"`

![image](https://github.com/eclipse-pde/eclipse.pde/assets/208716/49cdb6b6-83e8-49a0-8111-edf3e1672aa0)

The problem is that such features generate no p2 metadata and so for the EPP packages one cannot determine what the products require purely from the metadata.

This change generalize the support to always add such filtered metadata:

https://github.com/eclipse-equinox/p2/issues/423

It's highly likely you do not need this for your product...

**Comment by andrew-tram** [comment]
Ahh thank you for the info! That makes a lot more sense.

**Comment by laeubi** [comment]
Also note that this is [P2 feature](https://github.com/eclipse-equinox/p2), I'm not aware that PDE makes any use of this information when you launch a product.
